gramps/gen/filters/rules/_changedsincebase.py

```diff
@@ -93,16 +102,16 @@ def time_str_to_sec(self, time_str):
             )
         return time_sec
 
-    def prepare(self, db, user):
+    def prepare(self, db: Database, user):
         self.since = None
         self.before = None
         if self.list[0]:
             self.since = self.time_str_to_sec(self.list[0])
         if self.list[1]:
             self.before = self.time_str_to_sec(self.list[1])
 
-    def apply(self, db, obj):
-        obj_time = obj.get_change_time()
+    def apply_to_one(self, db: Database, obj: PrimaryObject) -> bool:
```

Comment:
For my own understanding why use the type `PrimaryObject` here when `TableObject` appears to be sufficient?

Reply:
We only have rules for the primary object types.  The `TableObject` includes the `Tag` object that is stored in a table, but is not considered a full primary object.  Table objects do contain a change timestamp though.

Reply:
I didn't spend a huge amount of time (but it took some time to get everything to pass the typing tests), but I also think that there may be a better way. In my tests, DataDict objects have no problem with any of the code. Thus, we might want to have a union type that could include objects with a Tag (in this particular case) or might be a Tag DataDict. I figure we can do lots more with types as we move forward.